analytics/src/test/kotlin/com/gravatar/analytics/tracks/TrackTrackerTest.kt

```diff
@@ -0,0 +1,39 @@
+package com.gravatar.analytics.tracks
+
+import com.automattic.android.tracks.TracksClient
+import com.gravatar.analytics.Event
+import io.mockk.mockk
+import io.mockk.verify
+import org.junit.Before
+import org.junit.Test
+
+class TrackTrackerTest {
+
+    private lateinit var tracker: TracksTracker
+    private lateinit var mockClient: TracksClient
+
+    @Before
+    fun setUp() {
+        mockClient = mockk<TracksClient>(relaxed = true)
+        tracker = TracksTracker(mockClient)
+    }
+
+    @Test
+    fun `when trackEvent is invoked then call track on client`() {
+        tracker.userId = "test_user_id"
+        val event = object : Event {
+            override val name: String = "test_event"
+        }
+
+        tracker.trackEvent(event)
+
+        verify { mockClient.track(event.name, "test_user_id", TracksClient.NosaraUserType.ANON) }
+    }
+
+    @Test
+    fun `when flush is invoked then call flush on client`() {
+        tracker.flush()
+
+        verify(exactly = 1) { mockClient.flush() }
```

Comment:
We could replace this with 
```suggestion
         verifySequence {
            mockClient.flush()
        }
```

This way, if the `TrackeClient.flush()` is modified in a way that it tracks events after flushing, the test will fail.

Reply:
🤔 Do you want to check if we track an event inside the `flush` method by mistake?

Reply:
Yeah, exactly. Just a suggestion, though.

Reply:
I don't think of this as a failure point, but I applied your suggestion anyway. (It's not going to hurt :) )